crates/bevy_ecs/src/system/query.rs

```diff
@@ -856,6 +858,178 @@ impl<'w, 's, D: QueryData, F: QueryFilter> Query<'w, 's, D, F> {
         }
     }
 
+    /// Returns the query item for the given [`Entity`].
```

Comment:
To avoid potential merge conflicts, let's try not to move these blocks of code around unnecessarily.